backend/src/main/java/edu/netcracker/backend/controller/ApproverCrudController.java

```diff
@@ -0,0 +1,68 @@
+package edu.netcracker.backend.controller;
+
+import edu.netcracker.backend.model.User;
+import edu.netcracker.backend.service.ApproverCrudService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class ApproverCrudController {
+
+    private ApproverCrudService acs;
+
+    @Autowired
+    public ApproverCrudController(ApproverCrudService acs) {
+        this.acs = acs;
+    }
+
+    @GetMapping("v1/api/admin/approvers/all")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN')")
+    public List<User> getAllApprovers() {
+       return acs.getAllApprovers();
+    }
+
+    @GetMapping("v1/api/admin/approvers/paging")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN')")
+    public List<User> getApprovers(@RequestParam("limit") Number limit, @RequestParam("offset") Number offset){
+        return acs.getApprovers(limit, offset);
+    }
+
+    @GetMapping("v1/api/admin/approvers")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN')")
+    public User getApprover(@RequestParam("id") Number id) {
+        Optional<User> user = acs.getById(id);
```

Comment:
It is better do work with Optional in the service layer. In controller you must work with User